app/org/sagebionetworks/bridge/models/CriteriaUtils.java

```diff
@@ -0,0 +1,116 @@
+package org.sagebionetworks.bridge.models;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Comparator;
+import java.util.Set;
+
+import org.springframework.validation.Errors;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.models.schedules.ScheduleContext;
+
+import com.google.common.base.Joiner;
+
+/**
+ * Utility classes for working with domain entities that should be matched by a growing list of 
+ * criteria, such as the version of the app making a request or the data groups associated to 
+ * a user. Matching is currently done through information passed in through the ScheduleContext, 
+ * a parameter object of values against which matching occurs.
+ */
+public class CriteriaUtils {
+    
+    public static Comparator<Criteria> SPECIFICITY_SORTER = new Comparator<Criteria>() {
```

Comment:
In-depth unit test for SPECIFICITY_SORTER?

Also, would this be better as a named inner class instead of an anonymous inner class?